internal/config/config.go

```diff
@@ -130,3 +104,179 @@ func RootGitRepoDir() (string, error) {
 func ProjectConfigFile() string {
 	return projectConfigName
 }
+
+func readAccessToken() (string, error) {
+	ring, err := openKeyring()
+
+	if errors.Is(err, keyring.ErrNoAvailImpl) {
+		accessToken, tokenErr := readAccessTokenPath()
+		return string(accessToken), tokenErr
+	}
+
+	item, err := ring.Get(keyringKey)
+	if err == nil {
+		return string(item.Data), nil
+	}
+
+	if errors.Is(err, keyring.ErrKeyNotFound) {
+		// Migrate to keychain
+		accessToken, tokenErr := readAccessTokenPath()
+		if len(accessToken) > 0 && tokenErr == nil {
+			return migrateAccessToken(ring, accessToken)
+		}
+		return "", nil
+	}
+
+	return "", err
+}
+
+func migrateAccessToken(ring keyring.Keyring, accessToken []byte) (string, error) {
+	err := ring.Set(keyring.Item{
+		Key:  keyringKey,
+		Data: accessToken,
+	})
+	if err != nil {
+		return "", err
+	}
+	path, err := accessTokenPath()
+	if err != nil {
+		return "", err
+	}
+	err = os.Remove(path)
+	if err != nil {
+		return "", err
+	}
+	return string(accessToken), nil
+}
+
+func WriteAccessToken(accessToken string) error {
+	ring, err := openKeyring()
+
+	if errors.Is(err, keyring.ErrNoAvailImpl) {
+		return writeAccessTokenPath(accessToken)
+	}
+
+	return ring.Set(keyring.Item{
+		Key:  keyringKey,
+		Data: []byte(accessToken),
+	})
+}
+
+func DeleteAccessToken() error {
+	ring, err := openKeyring()
+
+	if errors.Is(err, keyring.ErrNoAvailImpl) {
+		return deleteAccessTokenPath()
+	}
+
+	return ring.Remove(keyringKey)
+}
+
+func openKeyring() (keyring.Keyring, error) {
+	return keyring.Open(keyring.Config{
+		AllowedBackends: []keyring.BackendType{
+			keyring.SecretServiceBackend,
+			keyring.KWalletBackend,
+			keyring.KeychainBackend,
+			keyring.WinCredBackend,
+		},
+		ServiceName:              keyringService,
+		KeychainTrustApplication: true,
+		KeychainSynchronizable:   true,
+	})
+}
+
+func accessTokenPath() (string, error) {
+	dir, err := ConfigDir()
+	if err != nil {
+		return "", err
+	}
+
+	return path.Join(dir, keyringKey), nil
+}
+
+func readAccessTokenPath() ([]byte, error) {
+	var accessToken []byte
+	tokenPath, err := accessTokenPath()
+	if err != nil {
+		return nil, err
+	}
+
+	stat, err := os.Stat(tokenPath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			log.Fatal(err)
+		}
+		return nil, err
+	} else {
+		if stat.Mode()&^tokenFileMode != 0 {
+			err = os.Chmod(tokenPath, tokenFileMode)
+			if err != nil {
+				log.Printf("Unable to change %v file mode to 0%o: %v", tokenPath, tokenFileMode, err)
+			}
+		}
+		accessToken, err = ioutil.ReadFile(tokenPath)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
I think this is similar that it might be dependent on this working / treating a missing file as "no token configured".

That all could be improved, but I think that's something separate.